### PR TITLE
refactor(core): simplify snake/kebab case machinery and drop unused upper

### PR DESCRIPTION
**Pull request by wadakatu**
## Summary

`packages/core/src/utils/case.ts` still carried the ported case-conversion machinery (`unicodes`, `symbols`, a table of nine regexps, `prep`, `relax`, `fill`, `deapostrophe`, `lower`, `upper`), most of which only existed to serve `snake` / `kebab`. This replaces it with one `lower()` helper built from three literal character classes and four regexps, and removes `upper()`, which had no callers.

`pascal` / `camel` / `snake` / `kebab` / `conventionName` keep their exact behavior, including the quirks:
- acronyms are split (`getHTTPResponse` → `get_http_response`) only when the input has no separator characters
- all-caps input is lowercased first (`HTTP_RESPONSE` → `http_response`)
- apostrophes are dropped, leading separators become a leading fill character, trailing ones are removed

Net: -130 / +33 lines in `case.ts`. Follow-up to #4044 / #4110 / #4150.

## Verification

- Differential fuzz of old vs new `pascal`, `camel`, `snake`, `kebab`, `conventionName` over 1,428,589 unique inputs (every token in `tests/specifications` plus random strings drawn from ASCII, the Latin-1 ranges the regexps cover, apostrophes, whitespace, Greek and emoji): 0 differences
- `bun run test`: all packages pass, including 6 new `case.test.ts` cases pinning the rules above
- `tests/` snapshots regenerated: 7483 snapshots byte-identical
- `bun run build` and `bun run lint` (type-aware) pass

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved case conversion for acronyms, separators, apostrophes, leading or trailing whitespace, and empty values.
  * Standardized apostrophe handling across Pascal, camel, snake, and kebab case conversions.
  * Improved consistency when converting text containing symbols or mixed capitalization.

* **Tests**
  * Expanded coverage for case conversion edge cases, including acronyms, separators, apostrophes, whitespace, and empty strings.

### CODE REVIEW

**Review by melloware — APPROVED** [review]

File: packages/core/src/utils/case.ts [hunk 1]
```diff
@@ -126,20 +64,18 @@ export function pascal(s = '') {
...
+  return lower(s, '_');
 }
 
 export function kebab(s = '') {
-  return lower(s, '-', true);
-}
-
-export function upper(s: string, fillWith: string, isDeapostrophe?: boolean) {
-  return fill(up(prep(s, !!fillWith, false, true)), fillWith, isDeapostrophe);
+  return lower(s, '-');
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/core/src/utils/case.ts:78]: Per maintainer guidance on #4110 (unused `@orval/core` surface is removed without shims or a major bump), `upper` was dropped as-is. Merged by the maintainer.

File: packages/core/src/utils/case.ts [hunk 2]
```diff
@@ -126,20 +64,18 @@ export function pascal(s = '') {
...
 }
 
 export function conventionName(name: string, convention: NamingConvention) {
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/core/src/utils/case.ts:81]: Per maintainer guidance on #4110 (unused `@orval/core` surface is removed without shims or a major bump), `upper` was dropped as-is. Merged by the maintainer.
